### PR TITLE
catalog: add basic support for default and configured filters for entity page content

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Just the initial building blocks really. We had to remove the config schema from entity page contents as our config merging isn't built out enough yet.

The `isKind` and `isType` filter format is just *something* to get us started, not set in stone at all.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-react/alpha-api-report.md [hunk 1]
```diff
@@ -26,51 +24,60 @@ export function createEntityCardExtension<
   };
   disabled?: boolean;
   inputs?: TInputs;
-  configSchema?: PortableSchema<TConfig>;
```
Comment (camilaibs) [inline plugins/catalog-react/alpha-api-report.md:29]: Note: I'm using the config here, would you recommend I remove it for now?
https://github.com/backstage/backstage/pull/20720/files#diff-b5146cd23deb66381041855a8297ad170ab1d0eb49c81e4dfb97b2c4a197045f

File: plugins/catalog-react/src/alpha.tsx [hunk 2]
```diff
@@ -37,21 +36,57 @@ export { useEntityPermission } from './hooks/useEntityPermission';
...
+/** @alpha */
+export const entityFilterExtensionDataRef = createExtensionDataRef<
+  (ctx: { entity: Entity }) => boolean
+>('plugin.catalog.entity.filter');
+
+function applyFilter(a?: string, b?: string): boolean {
+  if (!a) {
+    return true;
+  }
+  return a.toLocaleLowerCase('en-US') === b?.toLocaleLowerCase('en-US');
```
Comment (camilaibs) [inline plugins/catalog-react/src/alpha.tsx:48]: Do you think typing errors might take time do catch? Adding a debug log would help in this case?

File: plugins/catalog-react/src/alpha.tsx [hunk 3]
```diff
@@ -37,21 +36,57 @@ export { useEntityPermission } from './hooks/useEntityPermission';
...
+  filterFunc?: (ctx: { entity: Entity }) => boolean,
+) {
+  return (ctx: { entity: Entity }) => {
+    const configuredFilterMatch = config.filter?.some(filter => {
+      const kindMatch = applyFilter(filter.isKind, ctx.entity.kind);
+      const typeMatch = applyFilter(
+        filter.isType,
+        ctx.entity.spec?.type?.toString(),
+      );
+      return kindMatch && typeMatch;
```
Comment (camilaibs) [inline plugins/catalog-react/src/alpha.tsx:64]: Do you this comparison could be hard to maintain depending on the number of filters we would like to apply? I was wondering if having a filter chain processor is worth it...

**Review by camilaibs — APPROVED** [review]
Left minor comments, nothing blocking

File: plugins/catalog-react/src/alpha.tsx (hunk 3 above)
Comment (Rugvip) [reply plugins/catalog-react/src/alpha.tsx:64]: Yep for sure, this is intended to be refactored into something sensible

File: plugins/catalog-react/alpha-api-report.md (hunk 1 above)
Comment (camilaibs) [reply plugins/catalog-react/alpha-api-report.md:29]: As we discussed, I'll remove the config from the referenced pull request for now.

**Pull request merged**